test/python/circuit/test_parameters.py

```diff
@@ -864,15 +864,17 @@ def test_expressions_of_parameter_with_constant(self):
 
         good_constants = [2, 1.3, 0, -1, -1.0, numpy.pi]
 
+        digit = 14
+
         x = Parameter('x')
 
         for op in self.supported_operations:
             for const in good_constants:
                 expr = op(const, x)
                 bound_expr = expr.bind({x: 2.3})
 
-                self.assertEqual(float(bound_expr),
-                                 op(const, 2.3))
+                self.assertEqual(round(float(bound_expr), digit),
+                                 round(op(const, 2.3), digit))
```

Comment:
Do these tests fail without the `round`? `resolve` doesn't change the final type so I'd be surprised to see that it introduces floating point errors here.

Reply:
rounding errors are sometimes happened in sympy.

```
import numpy
from sympy import *
from operator import *

x = Symbol('x')
c = numpy.pi

str_subs = "float(sympify('x + {0}'.format(c)).subs({x:2.3}))"
str_lambdify = "lambdify([x], '{0} + 2.3'.format(sympify(c)))(2.3)"
str_lambdify_with_simpify = "lambdify([x], '{0} + 2.3'.format(sympify(c)))(2.3)"

print ('{0}: {1}'.format(str_subs, eval(str_subs)))
print ('{0}: {1}'.format(str_lambdify, eval(str_lambdify)))
print ('{0}: {1}'.format(str_lambdify_with_simpify, eval(str_lambdify_with_simpify)))
```
output:
```
float(sympify('x + {0}'.format(c)).subs({x:2.3})): 5.441592653589793
lambdify([x], '{0} + 2.3'.format(c))(2.3): 5.441592653589793
lambdify([x], '{0} + 2.3'.format(sympify(c)))(2.3): 5.44159265358979
```